backend/wayland/output.c

```diff
@@ -26,56 +32,31 @@ static struct wl_callback_listener frame_listener;
 static void surface_frame_callback(void *data, struct wl_callback *cb,
 		uint32_t time) {
 	struct wlr_wl_output *output = data;
-	assert(output);
-	wl_callback_destroy(cb);
-	output->frame_callback = NULL;
-
-	wlr_output_send_frame(&output->wlr_output);
-}
-
-static struct wl_callback_listener frame_listener = {
-	.done = surface_frame_callback
-};
-
-static bool output_set_custom_mode(struct wlr_output *wlr_output,
-		int32_t width, int32_t height, int32_t refresh) {
-	struct wlr_wl_output *output = get_wl_output_from_output(wlr_output);
-	wl_egl_window_resize(output->egl_window, width, height, 0, 0);
-	wlr_output_update_custom_mode(&output->wlr_output, width, height, 0);
-	return true;
-}
 
-static bool output_make_current(struct wlr_output *wlr_output,
-		int *buffer_age) {
-	struct wlr_wl_output *output =
-		get_wl_output_from_output(wlr_output);
-	return wlr_egl_make_current(&output->backend->egl, output->egl_surface,
-		buffer_age);
-}
-
-static bool output_swap_buffers(struct wlr_output *wlr_output,
-		pixman_region32_t *damage) {
-	struct wlr_wl_output *output =
-		get_wl_output_from_output(wlr_output);
+	wl_callback_destroy(output->frame_callback);
 
-	if (output->frame_callback != NULL) {
-		wlr_log(WLR_ERROR, "Skipping buffer swap");
-		return false;
-	}
+	if (output->scheduled) {
+		struct wlr_wl_buffer *buffer = gbm_bo_get_user_data(output->scheduled);
```

Comment:
This is pretty weird. The frame event means "please render", not "please commit". Instead we should call `wlr_output_send_frame` here, and let the compositor swap buffers to commit.